docs/semgrep-ci/configuration-reference.md

```diff
@@ -22,124 +22,243 @@ Object.entries(frontMatter).filter(
-* Do **not** perform full scans on non-mainline or non-trunk branches. Performing full scans on every branch slows down your CI jobs, displays findings that developers did not introduce, and results in many duplicated findings in the Semgrep App, resulting in a poorer experience.
-:::
- 
-### Examples of `SEMGREP_BASELINE_REF`
+Set `SEMGREP_RULES` to define rules and rulesets for your scan. Findings are logged within your CI environment. `SEMGREP_RULES` is incompatible with `SEMGREP_APP_TOKEN`. Do not set `SEMGREP_APP_TOKEN` environment variable within the same CI job as `SEMGREP_RULES`.
 
-To only report findings newly added
-since branching off from your `main` branch, set the following:
-<pre class="language-bash"><code>SEMGREP_BASELINE_REF=main</code></pre>
+Examples:
 
-To only report findings newly added
-after a specific commit, set the following:
-<pre class="language-bash"><code>SEMGREP_BASELINE_REF=<span className="placeholder">INSERT_GIT_COMMIT_HASH</span></code></pre>
+```bash
+# Define a single ruleset:
+export SEMGREP_RULES="p/default"
 
-## Semgrep App token
+# Define multiple rule sources, delimited by a space: 
+export SEMGREP_RULES="p/default no-exec.yml"
+```
+
+### `SEMGREP_TIMEOUT`
 
-You can configure which rules to run also with Semgrep App.
-Get your token from [Semgrep App > Settings](https://semgrep.dev/manage/settings).
+Set `SEMGREP_TIMEOUT` to define a custom timeout. The value must be in seconds. The default value is 1800 seconds (30 minutes).
 
-<pre class="language-bash"><code>SEMGREP_APP_TOKEN=<span className="placeholder">TOKEN_VALUE</span></code></pre>
+Example:
 
-## Get hyperlinks in Semgrep App
+```bash
+export SEMGREP_TIMEOUT="900"
+```
 
-Set these variables to hyperlink to the correct repositories, files, and PRs
-in the Semgrep App UI & notifications.
+## Environment variables for creating hyperlinks in Semgrep App
 
-```sh
-SEMGREP_REPO_URL="https://github.com/foo/bar"
-SEMGREP_BRANCH="feature/add-new-bugs"
-SEMGREP_JOB_URL="https://ci-server.com/jobs/1234"
-SEMGREP_REPO_NAME="foo/bar"
-SEMGREP_COMMIT="a52bc1ef"
-SEMGREP_PR_ID="44"
+By default, Semgrep App autodetects values such as the name of your repository, which Semgrep uses to generate hyperlinks (URLs) to the specific repository code that generated the finding.
+
+Set any as needed or all of the following environment variables to troubleshoot and override autodetected CI environment values.
+
+### `SEMGREP_BRANCH`
+
+Set `SEMGREP_BRANCH` to define the branch name for the URL used to generate hyperlinks. To avoid hardcoding this value, check your CI provider's documentation for available environment variables that can automatically detect the correct values for every CI job. 
```

Comment:
Generate hyperlinks where? To display them on Findings page? I am not sure I understand what this option allows.

Reply:
Sure, I added in the link to the findings page.